moabb/datasets/liu2024.py

```diff
@@ -0,0 +1,310 @@
+    including preparation time and approximately 10 minutes of signal recording. Before the start of the MI experiment,
+    the patients opened their eyes and closed their eyes for 1 minute each. The MI experiment was divided into 40 trials, and
+    each trial took 8 seconds, which consisted of three stages (instruction, MI and break). In the instruction stage, patients
+    were prompted to imagine grasping a spherical object with the left- or right-hand. In the MI stage, participants imagined
+    performing this action, a video of gripping motion is played on the computer, which leads the patient imagine grabbing the
+    ball. This video stays playing for 4 s. Patients only imagine one hand movement.In the break stage, participants were allowed
+    to relax and rest. The MI experiments alternated between the left- and right-hand, and the patients moved onto the next stage
+    of the experiment according to the instructions.
+
+    The EEG data were collected through a wireless multichannel EEG acquisition system (ZhenTec NT1, Xi’an ZhenTec Intelligence
+    Technology Co., Ltd., China). The system includes an EEG cap, an EEG acquisition amplifier, a data receiver and host computer
+    software. The EEG cap had electrodes placed according to the international 10-10 system, including 29 EEG recording electrodes
+    and 2 electrooculography (EOG) electrodes. The reference electrode located at CPz position and the grounding electrode located
+    at FPz position. All the EEG electrodes and grounding electrode are Ag/AgCl semi-dry EEG electrodes based on highly absorbable
+    porous sponges that are dampened with 3% NaCl solution. The EOG electrodes are composed by Ag/AgCl electrodes and conductive
+    adhesive hydrogel. The common-mode rejection ratio was 120 dB, the input impedance was 1 GΩ, the input noise was less than
+    0.4 μVrms, and the resolution was 24 bits. The acquisition impedance was less than or equal to 20 kΩ. The sampling frequency
+    was 500 Hz.
+
+    In the dataset, we've removed the 2 EOG channels and we kept the 29 EEG channels and the STIM channel.
+
+    References
+    ----------
+
+    .. [1] Liu, Haijie; Lv, Xiaodong (2022). EEG datasets of stroke patients.
+        figshare. Dataset. DOI: https://doi.org/10.6084/m9.figshare.21679035.v5
+
+    .. [2] Liu, Haijie, Wei, P., Wang, H. et al. An EEG motor imagery dataset
+       for brain computer interface in acute stroke patients. Sci Data 11, 131
+       (2024). DOI: https://doi.org/10.1038/s41597-023-02787-8
+
+    Notes
+    -----
+
+    .. versionadded:: 1.1.1
+
+    """
+
+    def __init__(self):
+        super().__init__(
+            subjects=list(range(1, 50 + 1)),
+            sessions_per_subject=1,
+            events={"left_hand": 1, "right_hand": 2, "instr": 3, "break": 4},
+            code="Liu2024",
+            interval=(2, 6),
+            paradigm="imagery",
+            doi="10.1038/s41597-023-02787-8",
+        )
```

Comment:
I’m not sure if we should include the instruction and break events...
Because then by default, when using this dataset with the `MotorImagery` paradigm, algorithms would also have to classify these events. 
Plus, will the instructions and break epochs really correspond to these phases? i.e. 4s interval, starting 2s after the event.

I suggest we don’t include them by default. And eventually we can leave the possibility to get them with dataset arguments: 
```suggestion
    def __init__(self, break_events=False, instrumental_events=False):
        events = {"left_hand": 1, "right_hand": 2}
        if break_events:
            events[“instr”] = 3
        if insert_events:
            events[“break”] = 4
        super().__init__(
            subjects=list(range(1, 50 + 1)),
            sessions_per_subject=1,
            events=events,
            code="Liu2024",
            interval=(2, 6),
            paradigm="imagery",
            doi="10.1038/s41597-023-02787-8",
        )
```

Reply:
I added that possibility. If I understood correctly what you meant, I think the epochs will be respected since the duration and onset are respected in the events file as you can see the break and instructions last for exactly 4seconds after each stimulu: 
![image](https://github.com/NeuroTechX/moabb/assets/95372659/a473fa37-d1c2-4f10-9110-b63e5ac514da)

Reply:
The parameters `break_events` and `inst_events` are not working:
```python
data = Liu2024(break_events=False, instr_events=False).get_data([1])
data[1]['0']['0'].annotations
#  <Annotations | 120 segments: break (40), instr (40), left_hand (20), ...>
```
( you still get annotations with all possible events even if you set these parameters to False)

The rest is great :)